### PR TITLE
8254173: Add Zero, Minimal hotspot targets to submit workflow

### PR DESCRIPTION
**Pull request by shipilev**
Zero VM and Minimal VM builds are routinely discovering the problems with internal Hotspot dependencies. Mostly because they turn off the whole lot of VM features, and every path that is not guarded by a feature #ifdef or build file list fails. 

It would be good to add Zero and Minimal targets to submit workflow.

It seems to add two ~9 minute stages, compared to ~17 minute stage for building no-pch hotspot, and ~33 minutes for full JDK.

Attention @rwestberg.

Testing:
  - [x] GH workflow still works; see [latest run ](https://github.com/shipilev/jdk/actions/runs/293805791)

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- |
| Build | ⏳ (5/5 running) | ⏳ (2/2 running) | ⏳ (2/2 running) |

### Issue
 * [JDK-8254173](https://bugs.openjdk.java.net/browse/JDK-8254173): Add Zero, Minimal hotspot targets to submit workflow

### Reviewers
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to 7e3e6ad92d1db908ae18ae27f19ca49b274d9e2a
 * [Robin Westberg](https://openjdk.java.net/census#rwestberg) (@rwestberg - Committer)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/546/head:pull/546`
`$ git checkout pull/546`

### CODE REVIEW

**Review by erikj79 — APPROVED** [review]

**Review by rwestberg — APPROVED** [review]
Looks good to me! Adding a few additional shorter build shouldn't increase the total runtime, as the test execution waits for all builds to be done, so we are limited by the full builds anyway.

One minor comment, perhaps rearrange the description a bit of these new flavors with the differentiating parts first. GitHub cuts them short in the overview list (see the list on the left at https://github.com/shipilev/jdk/actions/runs/293805791 for an example) so you can't currently tell which one is which in the overview without clicking on them.

File: .github/workflows/submit.yml [hunk 1]
```diff
@@ -106,13 +106,21 @@ jobs:
           - build release
           - build debug
           - build hotspot no-pch
+          - build debug hotspot no-pch zero
```
[the remark this answers is not in the document]
Comment (shipilev) [reply .github/workflows/submit.yml:109]: That makes sense. We should change the order in #547 then first? So the final thing would be:

```
- build release
- build debug
- build hotspot no-pch debug
- build hotspot zero no-pch debug
- build hotspot minimal no-pch debug
```

[the remark this answers is not in the document]
Comment (shipilev) [reply .github/workflows/submit.yml:109]: ...or we drop `debug` to altogether:

```
- build release
- build debug
- build hotspot no-pch
- build hotspot zero no-pch
- build hotspot minimal no-pch
```

[the remark this answers is not in the document]
Comment (rwestberg) [reply .github/workflows/submit.yml:109]: Yeah I had the same comment there. :) Perhaps just drop "debug" qualifier as they all have it. And perhaps drop the no-pch from the two new ones, it's IMO not that important to display it in the description (unless we were to add for example zero without no-pch) - the full config is fairly easy to figure out from the yml or the test log anyway.

**Review by rwestberg — APPROVED** [review]
Looks good, thanks for fixing!

**Pull request closed without merging**